core/src/main/java/org/apache/calcite/sql/fun/SqlItemOperator.java

```diff
@@ -89,6 +91,17 @@
     }
     final RelDataType operandType = callBinding.getOperandType(0);
     final SqlSingleOperandTypeChecker checker = getChecker(operandType);
+    if (checker == null) {
+      SqlNode node = callBinding.getCall()
```

Comment:
There is a util method for generating readable signature error:
https://github.com/apache/calcite/blob/986a2d579c8f9b9f08aa9bbbfe11efc4e7bb0809/core/src/main/java/org/apache/calcite/sql/SqlCallBinding.java#L279-L284
Is it enough to use that method rather than manually create new error messages?

Reply:
Thanks hongze! The `newValidationSignatureError()` can solve the problem.